src/plugins/shared_ux/public/components/page_template/no_data_page/no_data_card/no_data_card.tsx

```diff
@@ -24,12 +24,16 @@ export const NoDataCard: FunctionComponent<NoDataCardProps> = ({
   title,
   button,
   description,
+  renderFooter,
   ...cardRest
 }) => {
   const footer = () => {
     if (typeof button !== 'string') {
       return button;
     }
+    if (!button && !renderFooter) {
```

Comment:
I introduced this new prop, because the check differs a bit between `no_data_card` and `elastic_agent_card`. In `elastic_agent_card`, if no button was provided, we'd render the default button:
https://github.com/elastic/kibana/blob/main/src/plugins/kibana_react/public/page_template/no_data_page/no_data_card/elastic_agent_card.tsx#L71

Whereas in `no_data_card`, if no button is present, we render nothing;
https://github.com/elastic/kibana/blob/main/src/plugins/kibana_react/public/page_template/no_data_page/no_data_card/no_data_card.tsx#L25

I don't know if this logic was intentional, but with this check it will keep consistent to what it was.